bok_choy/browser.py

```diff
@@ -207,7 +207,7 @@ def browser_check_func():
             return False, None
 
     browser_instance = Promise(
-        browser_check_func, "Browser is instantiated successfully.", try_limit=3).fulfill()
+        browser_check_func, "Browser is instantiated successfully.", try_limit=3, timeout=60).fulfill()
```

Comment:
Ugh, yeah it looks like bok-choy is a bit clunky/confusing when you use try_limit, because it will stop [when the first of the limits (tries or time) is reached](https://github.com/edx/bok-choy/blob/master/bok_choy/promise.py#L113). :(

With an underlying 30-second polling function, with the timeout=60 I'm not sure that you will actually try 3 times either. I think it would be better to set this value even higher (120?) so that we are sure that the try_limit is the one that is expiring first. Together with a comment here that the try_limit, rather than the timeout value, is what we are expecting to kick in.

As an aside, I do think that this confusing bok-choy behavior is a bug. If you only specify try_limit I would assume that your intention is to actually use the value you gave it, not a mysterious default timeout. :(  We should discuss in an issue, because if we change the behavior now it could cause existing tests (not just ours, as edX is not the only org using bok-choy) to fail. Perhaps tho, we **should** change it before we get to v1.0 which hopefully will be soon.